src/main/java/backend/RepoId.java

```diff
@@ -0,0 +1,68 @@
+package backend;
+
+import org.eclipse.egit.github.core.RepositoryId;
+import util.Utility;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoId {
+
+    private final String repoOwner;
+    private final String repoName;
+
+    public RepoId(String repoIdString) throws IllegalArgumentException {
+        throwExceptionIfNotWellFormed(repoIdString);
+
+        String[] repoIDComponents = repoIdString.toLowerCase().split("/");
+        this.repoOwner = Utility.removeAllWhitespace(repoIDComponents[0]);
+        this.repoName = Utility.removeAllWhitespace(repoIDComponents[1]);
+    }
+
+    /**
+     * Checks if repoIdString non-null, non-empty
+     * and is of form <repoOwner>/<repoName>
+     * @param repoIdString
+     * @return True if the above conditions are met else False is returned
+     */
+    public static boolean isWellFormedRepoIdString(String repoIdString) {
+        RepositoryId repositoryId = RepositoryId.createFromId(repoIdString);
```

Comment:
These `isWellFormed` methods delegate to EGit's repo id class to check validity, parse, etc. The intention of creating this class was partially to decouple us from the API code, so I guess we should implement our own checks.